core/i18n/src/ITwinLocalization.ts

```diff
@@ -117,10 +123,16 @@ export class ITwinLocalization implements Localization {
    * @throws Error if no keys resolve to a string.
    * @public
    */
-  public getLocalizedString(key: string | string[], options?: TOptionsBase): string {
+  public getLocalizedString(key: string | string[], options?: TranslationOptions): string {
+    if (options?.returnDetails || options?.returnObjects) {
```

Comment:
if thats the case, why dont we narrow the type of the second arg to omit those properties?

Reply:
From i18next, `TOptionsBase` allows returnDetails and returnObjects keys. Our `TranslationOptions` does not define them, but it technically allows any key-value pair to be passed in. Since `getLocalizedString()` specifically should be returning a string, I added this check so that it won't try to return an object.